### DISCUSSION TITLE
knot-resolver нагружает сервер

### DISCUSSION THREAD

**Discussion by Murakumo-JP** [xtrime-ru/antizapret-vpn-docker#105] [Q&A] [answered]
Здраствуйте, помогите, после того как стал использовать  AmneziaWG + Adguard стал замечать что раз в пару дней да и днём пару раз может быть такое что knot-resolver нагружает сервер под 100% загрузка идёт на CPU и на память, на ntc.party вроде писали что кеш можно почистить но я как понял команда `echo "cache.clear()" | socat - /run/knot-resolver/control/1` не работает под Ubuntu 24.04, ну или как то может по другому можно решить эту проблему ? я уже и систему сносил и всё по новой настраивал и один фиг повторяется то же самое.

Нагрузку на сервер даёт вот это 
![image](https://github.com/user-attachments/assets/1b908561-cb81-4dfc-96ea-ecc8ff9ed6b9)

**Comment by Ernieleo** [comment]
Напишите хотя бы конфигурацию сервера, сколько всего RAM хотя бы.

**Reply by Murakumo-JP** [reply]
Обычный самый дешёвый сервер
1 ядро CPU
1 гб оперативы 
10 гб ssd вроде там.
knot-resolver грузит сервер раз в пару дней или раз в день и после перезагрузки сервера, когда использовал чисто OpenVPN такой проблемы не была.

**Comment by xtrime-ru** [comment]
Кеш не причем. Нужно проверить что сервер не уходит в swap. Если уходит, то нужно больше оперативной памяти. Дайте полный скриншот команды top во время проблем с нагрузкой.

**Reply by MascotWorld** [reply]
![image](https://github.com/user-attachments/assets/4ae408da-bc6b-4ef0-8c70-cf0f824ddd9a)
такая же проблема, сервер такой же по конфигурации

**Reply by xtrime-ru** [reply]
Либо менять сервер, либо выключать загрузку списков и формировать список только из нужных сайтов вручную через include файлы в папке config.

**Reply by MascotWorld** [reply]
а раз в какое время он обновляет данные? Если из-за обновления данных, можно отключить автообновление и обновлять только при старте контейнера. Есть возможность как-то выключить?

**Reply by xtrime-ru** [reply, marked as answer]
Установить ENV переменную UPDATE_TIMER=3660d равной 10 годам например.

**Reply by Murakumo-JP** [reply]
Ой я и забыл про вопрос этот, спасибо за совет, поставил 30 дней не знаю будет работать или нет но гляну, спасибо.
